### PR TITLE
gotest feature and fix from service team

### PR DESCRIPTION
**Pull request by tadelesh**
resolve: https://github.com/Azure/autorest.go/issues/1193
fix: https://github.com/Azure/autorest.go/issues/1203

### CODE REVIEW

File: packages/autorest.gotest/src/template/exampleTest.go.njk [hunk 1]
```diff
@@ -17,8 +17,8 @@ import (
 {%- set allExamples = [] %}
 {%- for exampleGroup in exampleGroups %}
 {%- for example in exampleGroup.examples %}
-{%- if allExamples.indexOf(example.originalFile)<0 %}
```
Comment (raych1) [inline packages/autorest.gotest/src/template/exampleTest.go.njk:20]: What's the difference between `originalFile` and `name`?

File: packages/autorest.gotest/src/generator/goLinter.ts [hunk 2]
```diff
@@ -3,20 +3,22 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import * as path from 'path';
 import { AutorestExtensionHost } from '@autorest/extension-base';
-import { Config, configDefaults } from '../common/constant';
-import { Helper } from '@autorest/testmodeler/dist/src/util/helper';
 import { TestConfig } from '@autorest/testmodeler/dist/src/common/testConfig';
+import { Helper } from '@autorest/testmodeler/dist/src/util/helper';
```
Comment (raych1) [inline packages/autorest.gotest/src/generator/goLinter.ts:8]: Seems we don't have lint rule enabled in this project?

**Review by raych1 — COMMENTED** [review]
Does the CI failure matter?

File: packages/autorest.gotest/src/template/exampleTest.go.njk (hunk 1 above)
Comment (tadelesh) [reply packages/autorest.gotest/src/template/exampleTest.go.njk:20]: `name` will always be there, but `originalFile` will be empty if `include-x-ms-examples-original-file` set to false.

File: packages/autorest.gotest/src/generator/goLinter.ts (hunk 2 above)
Comment (tadelesh) [reply packages/autorest.gotest/src/generator/goLinter.ts:8]: what do you mean lint rule?

**Review by raych1 — APPROVED** [review]

**Pull request merged**